### PR TITLE
Allow initializing `BitVec` and `AtomicBitVec` with a custom value

### PR DESCRIPTION
**Pull request by MatteoH2O1999**
This PR adds the ability to initialize `BitVec` and `AtomicBitVec` with a custom `bool` value instead of always defaulting to `false`.

### CODE REVIEW

File: src/bits/bit_vec.rs [hunk 1]
```diff
@@ -82,13 +82,22 @@ impl<B: AsRef<[usize]>> Index<usize> for BitVec<B> {
 }
 
 impl BitVec<Vec<usize>> {
-    /// Create a new bit vector of length `len`.
+    /// Create a new bit vector of length `len` initialized to `false`.
```
Comment (vigna) [inline src/bits/bit_vec.rs:85]: Guidelines: documentation is in third person ("Creates a new bit vector...").

Comment (MatteoH2O1999) [reply src/bits/bit_vec.rs]: Resolved

**Comment by vigna** [comment]
It might be interesting to have a `vec!`-like macro, so that you can do `bitvec![100; false]` or  `bitvec![100; true]`  but also `bitvec![true, true, true, false]` or `bitvec![0, 1, 0, 1]`.

**Comment by vigna** [comment]
You can have a look at the `vec!` macro, it's pretty simple.

**Comment by vigna** [comment]
We don't need that for the atomic case because there's `convert_to`.

**Pull request merged**